### PR TITLE
Add evaluate v1

### PR DESCRIPTION
**Pull request by abhi1thakur**

### CODE REVIEW

File: src/autonlp/cli/evaluate.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+{BLD}`entity_extraction`{RST}:
+    {BLD}col_name1{RST} -> {BLD}tokens{RST}  (The tokens to tag)
+    {BLD}col_name2{RST} -> {BLD}tags{RST}    (The associated tags)
+    Example col_mapping: --col_mapping '{GRN}col_name1{RST}:{CYN}tokens{RST},{GRN}col_name2{RST}:{CYN}tags{RST}'
+
+{BLD}`speech_recognition`{RST}:
+    {BLD}col_name1{RST} -> {BLD}path{RST}  (The path to the audio file, only the file name matters)
+    {BLD}col_name2{RST} -> {BLD}text{RST}  (The matching speech transcription)
+    Example col_mapping: --col_mapping '{GRN}col_name1{RST}:{CYN}path{RST},{GRN}col_name2{RST}:{CYN}text{RST}'
+"""
```
Comment (SBrandeis) [inline src/autonlp/cli/evaluate.py:38]: Let's put this in some common place and reuse it instead of copy-pasting

File: src/autonlp/autonlp.py [hunk 2]
```diff
@@ -99,6 +100,23 @@ def create_project(self, name: str, task: str, language: str, max_models: int):
...
+        payload = {
+            "username": self.username,
+            "dataset": dataset,
+            "task": task_id,
+            "model": model,
+            "col_mapping": col_mapping,
+            "subset": subset,
+        }
+        json_resp = http_post(path="/evaluate/create", payload=payload, token=self.token).json()
+        self._eval_proj = Evaluate.from_json_resp(json_resp, token=self.token)
```
Comment (SBrandeis) [inline src/autonlp/autonlp.py:117]: Let's add this in __init__ method:
```python
self._eval_proj = None
```

File: src/autonlp/cli/evaluate.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            default=None,
+            required=True,
+            help=COL_MAPPING_HELP,
+        )
+        create_evaluation_parser.add_argument(
+            "--subset",
+            type=str,
+            default="test",
+            required=False,
+            help="Which subset of dataset to use for evaluation. If not provided, this will default to 'test'.",
```
Comment (SBrandeis) [inline src/autonlp/cli/evaluate.py:92]: IMO "split" is better than "subset", as it matches `datasets` wording

File: src/autonlp/evaluate.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+
+JOB_STATUS = (
+    ("⌚", "queued"),
+    ("🚀", "start"),
+    ("⚙", "data_munging"),
+    ("🏃", "model_evaluating"),
+    ("✅", "success"),
+    ("❌", "failed"),
+)
```
Comment (SBrandeis) [inline src/autonlp/evaluate.py:16]: Let's share this between training and eval jobs

Comment (SBrandeis) [reply src/autonlp/evaluate.py:16]: I think it makes sense to move the Evaluate class to the `model.py` file

File: src/autonlp/evaluate.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        if value == task_id:
+            return key
+    return "❌ Unsupported task! Please update autonlp"
+
+
+def get_eval_job_status(status_id: int) -> Tuple[str, str]:
+    try:
+        return JOB_STATUS[status_id - 1]
+    except IndexError:
+        return "❓", "Unhandled status! Please update autonlp"
```
Comment (SBrandeis) [inline src/autonlp/evaluate.py:30]: This is from `project`, let's move them to a `utils` package and reuse them.

Please avoid copy-pasting things that don't need to! It's better to have a single piece of code to manage

**Review by SBrandeis — CHANGES_REQUESTED** [review]
See comments below

File: src/autonlp/evaluate.py (hunk 4 above)
Comment (abhi1thakur) [reply src/autonlp/evaluate.py:16]: removed for time being

File: src/autonlp/autonlp.py [hunk 6]
```diff
@@ -111,7 +112,7 @@ def create_evaluation(self, task: str, dataset: str, model: str, col_mapping: st
             "task": task_id,
             "model": model,
             "col_mapping": col_mapping,
```
Comment (SBrandeis) [inline src/autonlp/autonlp.py:114]: `col_mapping` should be sent as a dict here

File: src/autonlp/evaluate.py [hunk 7]
```diff
@@ -46,13 +18,10 @@ class Evaluate:
     @classmethod
     def from_json_resp(cls, json_resp: dict, token: str):
         task = get_task(json_resp["task"])
-        status_emoji, status = get_eval_job_status(json_resp["status"])
         return cls(
             _token=token,
             user=json_resp["username"],
             task=task,
-            status_emoji=status_emoji,
-            status=status,
```
Comment (SBrandeis) [inline src/autonlp/evaluate.py]: No status reporting anymore ?

**Review by SBrandeis — CHANGES_REQUESTED** [review]
Great, thanks! Just a couple of other changes before we merge

File: src/autonlp/evaluate.py (hunk 7 above)
Comment (abhi1thakur) [reply src/autonlp/evaluate.py]: not really. i will add it with some status checking cli command in near future

**Pull request merged**